examples/ecs/observers.rs

```diff
@@ -0,0 +1,199 @@
+//! This examples illustrates several ways you can employ observers
+
+use bevy::{
+    prelude::*,
+    utils::{HashMap, HashSet},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (draw_shapes, handle_click))
+        .run();
+}
+
+#[derive(Component)]
+struct Mine {
+    pos: Vec2,
+    size: f32,
+}
+
+#[derive(Component)]
+struct TriggerMines {
+    pos: Vec2,
+    radius: f32,
+}
+
+#[derive(Component)]
+struct Explode;
+
+fn setup(world: &mut World) {
+    world.spawn(Camera2dBundle::default());
+
+    let font = world
+        .resource::<AssetServer>()
+        .load("fonts/FiraMono-Medium.ttf");
+    world.spawn(TextBundle::from_section(
+        "Click on a \"Mine\" to trigger it.\n\
+            When it explodes it will trigger all overlapping mines.",
+        TextStyle {
+            font,
+            font_size: 24.,
+            color: Color::WHITE,
+        },
+    ));
+
+    // Pre-register all our components, resource and event types.
+    world.init_resource::<SpatialIndex>();
+    world.init_component::<Mine>();
+    world.init_component::<TriggerMines>();
+    world.init_component::<Explode>();
+
+    // Observers are triggered when a certain event is fired, each event is represented by a component type.
```

Comment:
The confusion between events (sensu observers) and Bevy's existing events is really strong here.